### PR TITLE
[ML] Update documentation regarding versioning.

### PR DESCRIPTION
**Pull request by edsavage**
Update the ml and transform reference documentation to provide information regarding the new versioning schemes independent from the product versions.

### CODE REVIEW

File: docs/reference/ml/ml-shared.asciidoc [hunk 1]
```diff
@@ -419,6 +419,10 @@ of the most recent snapshot for this job. Valid values range from `0` to
 jobs created before version 7.8.0, the default value matches
 `model_snapshot_retention_days`. For more information, refer to
 {ml-docs}/ml-ad-run-jobs.html#ml-ad-model-snapshots[Model snapshots].
+Note that as of {es} version 8.10.0 a new version number has been introduced to
+track changes to configuration and state in the machine learning plugin. This
```
Comment (droberts195) [inline docs/reference/ml/ml-shared.asciidoc:423]:
```suggestion
track changes to configuration and state in the {ml} plugin. This
```

File: docs/reference/ml/anomaly-detection/apis/upgrade-job-model-snapshot.asciidoc [hunk 2]
```diff
@@ -6,6 +6,10 @@
 ++++
 
 Upgrades an {anomaly-detect} model snapshot to the latest major version.
+Note that as of {es} version 8.10.0 a new version number has been introduced to
+track changes to configuration and state in the machine learning plugin. This
```
Comment (droberts195) [inline docs/reference/ml/anomaly-detection/apis/upgrade-job-model-snapshot.asciidoc:10]:
```suggestion
track changes to configuration and state in the {ml} plugin. This
```

File: docs/reference/ml/anomaly-detection/apis/get-snapshot.asciidoc [hunk 3]
```diff
@@ -106,6 +106,11 @@ properties:
 
 `min_version`::
 (string) The minimum version required to be able to restore the model snapshot.
+Note that as of {es} version 8.10.0 a new version number has been introduced to
+track changes to configuration and state in the machine learning plugin. This
```
Comment (droberts195) [inline docs/reference/ml/anomaly-detection/apis/get-snapshot.asciidoc:110]:
```suggestion
track changes to configuration and state in the {ml} plugin. This
```

File: docs/reference/ml/anomaly-detection/apis/get-job.asciidoc [hunk 4]
```diff
@@ -210,6 +210,10 @@ value.
 
 `job_version`::
 (string) The version of {es} that existed on the node when the job was created.
+Note that as of {es} version 8.10.0 a new version number has been introduced to
+track changes to configuration and state in the machine learning plugin. This
```
Comment (droberts195) [inline docs/reference/ml/anomaly-detection/apis/get-job.asciidoc:214]:
```suggestion
track changes to configuration and state in the {ml} plugin. This
```

File: docs/reference/ml/ml-shared.asciidoc [hunk 5]
```diff
@@ -1410,6 +1414,10 @@ the cluster, capped at `1024mb`. The default value is relatively small to
 ensure that high resource usage is a conscious decision. If you have jobs that
 are expected to analyze high cardinality fields, you will likely need to use a
 higher value.
+Note that as of {es} version 8.10.0 a new version number has been introduced to
+track changes to configuration and state in the machine learning plugin. This
```
Comment (droberts195) [inline docs/reference/ml/ml-shared.asciidoc:1418]:
```suggestion
track changes to configuration and state in the {ml} plugin. This
```

File: docs/reference/ml/ml-shared.asciidoc [hunk 6]
```diff
@@ -1856,6 +1864,10 @@ tokenizer in old versions of the product (before 6.2). `ml_classic` was
 the default categorization tokenizer in versions 6.2 to 7.13, so if you
 need categorization identical to the default for jobs created in these
 versions, specify `"tokenizer": "ml_classic"` in your `categorization_analyzer`.
+Note that as of {es} version 8.10.0 a new version number has been introduced to
+track changes to configuration and state in the machine learning plugin. This
```
Comment (droberts195) [inline docs/reference/ml/ml-shared.asciidoc:1868]:
```suggestion
track changes to configuration and state in the {ml} plugin. This
```

File: docs/reference/ml/trained-models/apis/get-trained-models.asciidoc [hunk 7]
```diff
@@ -1430,6 +1430,10 @@ A comma delimited string of tags. A trained model can have many tags, or none.
 `version`::
 (string)
 The {es} version number in which the trained model was created.
+Note that as of {es} version 8.10.0 a new version number has been introduced to
+track changes to configuration and state in the machine learning plugin. This
```
Comment (droberts195) [inline docs/reference/ml/trained-models/apis/get-trained-models.asciidoc:1434]:
```suggestion
track changes to configuration and state in the {ml} plugin. This
```

File: docs/reference/ml/trained-models/apis/put-trained-models.asciidoc [hunk 8]
```diff
@@ -13,6 +13,12 @@ WARNING: Models created in version 7.8.0 are not backwards compatible
          all nodes must be at least 7.8.0 to use a model stored by
          a 7.8.0 node.
 
+WARNING: As of product version 8.10.0 a new version number has been
+         introduced to track changes to configuration and state in
+         the machine learning plugin. This new version number is
+         decoupled from the product version and will increment
+         independently as necessary.
```
Comment (droberts195) [inline docs/reference/ml/trained-models/apis/put-trained-models.asciidoc:20]: I don't think this warning is necessary here. A user putting a new model does not need to know this, and certainly not as a warning.

File: docs/reference/transform/apis/get-transform.asciidoc [hunk 9]
```diff
@@ -78,7 +78,10 @@ This property is informational; you cannot change its value.
 
 `version`::
 (string) The version of {es} that existed on the node when the {transform} was
-created.
+created. Note that as of {es} version 8.10.0 a new version number has been introduced to
+track changes to configuration and state in the machine learning plugin. This
```
Comment (droberts195) [inline docs/reference/transform/apis/get-transform.asciidoc:82]:
```suggestion
track changes to configuration and state in the {transform} plugin. This
```

File: docs/reference/transform/apis/upgrade-transforms.asciidoc [hunk 10]
```diff
@@ -26,30 +26,34 @@ Requires the following privileges:
...
-configuration format and upgrades them to the latest version; including clean up 
-of the internal data structures that store {transform} state and checkpoints. 
+{transforms-cap} are compatible across minor versions and between supported
+major versions. However, over time, the format of {transform} configuration
+information may change. This API identifies {transforms} which have a legacy
+configuration format and upgrades them to the latest version; including clean up
+of the internal data structures that store {transform} state and checkpoints.
 {transform-cap} upgrade does not affect the source and destination indices.
+Note that as of {es} version 8.10.0 a new version number has been introduced to
+track changes to configuration and state in the transform plugin. This
```
Comment (droberts195) [inline docs/reference/transform/apis/upgrade-transforms.asciidoc:36]:
```suggestion
track changes to configuration and state in the {transform} plugin. This
```

File: docs/reference/transform/limitations.asciidoc [hunk 11]
```diff
@@ -6,9 +6,12 @@
...
 
-The following limitations and known problems apply to the {version} release of 
-the Elastic {transform} feature. The limitations are grouped into the following 
-categories:
+The following limitations and known problems apply to the {version} release of
+the Elastic {transform} feature (Note that as of {es} version 8.10.0 a new
+version number has been introduced to track changes to configuration and state
+in the transform plugin. This new version number is decoupled from the
+product version and will incrementindependently as necessary).
+The limitations are grouped into the following categories:
```
Comment (droberts195) [inline docs/reference/transform/limitations.asciidoc:14]: I don't think we should say decoupling the versions is a limitation. We've done it in a transparent way such that most users don't need to care. So please revert this file to the way it was before.

File: docs/reference/transform/apis/get-transform.asciidoc [hunk 12]
```diff
@@ -78,7 +78,10 @@ This property is informational; you cannot change its value.
 
 `version`::
 (string) The version of {es} that existed on the node when the {transform} was
-created.
+created. Note that as of {es} version 8.10.0 a new version number has been introduced to
+track changes to configuration and state in the {transform} plugin. This
+new version number is decoupled from the product version and will increment
+independently as necessary.
```
Comment (szabosteve) [inline docs/reference/transform/apis/get-transform.asciidoc:84]: As above: the parameter description might also need to be updated.

```suggestion
created.

NOTE: From {es} 8.10.0,  a new version number is used to
track the configuration and state changes in the {ml} plugin. This new 
version number is decoupled from the product version and will increment
independently. The `version` value represents the new version number.
```

File: docs/reference/transform/apis/upgrade-transforms.asciidoc [hunk 13]
```diff
@@ -26,30 +26,34 @@ Requires the following privileges:
...
+{transforms-cap} are compatible across minor versions and between supported
+major versions. However, over time, the format of {transform} configuration
+information may change. This API identifies {transforms} which have a legacy
+configuration format and upgrades them to the latest version; including clean up
+of the internal data structures that store {transform} state and checkpoints.
 {transform-cap} upgrade does not affect the source and destination indices.
+Note that as of {es} version 8.10.0 a new version number has been introduced to
+track changes to configuration and state in the {transform} plugin. This
+new version number is decoupled from the product version and will increment
+independently as necessary.
```
Comment (szabosteve) [inline docs/reference/transform/apis/upgrade-transforms.asciidoc:38]:
```suggestion

NOTE: From {es} 8.10.0,  a new version number is used to
track the configuration and state changes in the {ml} plugin. This new 
version number is decoupled from the product version and will increment
independently.
```

**Review by szabosteve — COMMENTED** [review]
I recommend adding the new info to a NOTE admonition.
In some cases, I _think_ the parameter descriptions might also need to be updated. I left a couple of review messages in the respective places.

File: docs/reference/ml/anomaly-detection/apis/get-job.asciidoc [hunk 14]
```diff
@@ -209,7 +209,11 @@ value.
 (string) Reserved for future use, currently set to `anomaly_detector`.
 
 `job_version`::
-(string) The version of {es} that existed on the node when the job was created.
+(string) The {ml} configuration version number at which the the job was created.
+NOTE: From {es} 8.10.0,  a new version number is used to
+track the configuration and state changes in the {ml} plugin. This new
+version number is decoupled from the product version and will increment
+independently. The `job_version` value represents the new version number.
```
Comment (szabosteve) [inline docs/reference/ml/anomaly-detection/apis/get-job.asciidoc:216]:
```suggestion

NOTE: From {es} 8.10.0,  a new version number is used to
track the configuration and state changes in the {ml} plugin. This new
version number is decoupled from the product version and will increment
independently. The `job_version` value represents the new version number.
```

File: docs/reference/ml/anomaly-detection/apis/get-snapshot.asciidoc [hunk 15]
```diff
@@ -105,7 +105,13 @@ properties:
...
 
 `min_version`::
-(string) The minimum version required to be able to restore the model snapshot.
+(string) The minimum {ml} configuration version number required to be
+able to restore the model snapshot.
+NOTE: From {es} 8.10.0,  a new version number is used to
+track the configuration and state changes in the {ml} plugin. This new
+version number is decoupled from the product version and will increment
+independently. The `min_version` value represents the new version number.
+
```
Comment (szabosteve) [inline docs/reference/ml/anomaly-detection/apis/get-snapshot.asciidoc:114]:
```suggestion

NOTE: From {es} 8.10.0,  a new version number is used to
track the configuration and state changes in the {ml} plugin. This new
version number is decoupled from the product version and will increment
independently. The `min_version` value represents the new version number.
```

File: docs/reference/ml/ml-shared.asciidoc [hunk 16]
```diff
@@ -1410,6 +1414,11 @@ the cluster, capped at `1024mb`. The default value is relatively small to
 ensure that high resource usage is a conscious decision. If you have jobs that
 are expected to analyze high cardinality fields, you will likely need to use a
 higher value.
+
+NOTE: From {es} 8.10.0,  a new version number is used to
+track the configuration and state changes in the {ml} plugin. This new
+version number is decoupled from the product version and will increment
+independently.
```
Comment (szabosteve) [inline docs/reference/ml/ml-shared.asciidoc:1421]:
```suggestion
+
--
NOTE: From {es} 8.10.0,  a new version number is used to
track the configuration and state changes in the {ml} plugin. This new
version number is decoupled from the product version and will increment
independently.
--
```

**Review by szabosteve — APPROVED** [review]
Sorry, asciidoc syntax strikes again. Left a couple of comments to fix it. If you don't have the time to deal with them, just merge the PR and I'll open another to address the syntax. Thank you! LGTM!

**Pull request merged**